x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -54,15 +54,6 @@ public record TestCaseSupplier(String name, List<DataType> types, Supplier<TestC
         Supplier<TestCaseSupplier.TestCase> {
 
     private static final Logger logger = LogManager.getLogger(TestCaseSupplier.class);
-    /**
-     * Build a test case without types.
-     *
-     * @deprecated Supply types
-     */
-    @Deprecated
-    public TestCaseSupplier(String name, Supplier<TestCase> supplier) {
-        this(name, null, supplier);
-    }
```

Comment:
After the bulk of this PR there were only four calls to this deprecated method so I migrated those too and zapped this!